### PR TITLE
Add sklearn metrics

### PR DESCRIPTION
**Pull request by giulio-tirabassi**
Example of classification plot with accuracy
immagine

### CODE REVIEW

File: omigami/models/sklearn_metrics.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        jaccard_score, pos_label=None, average="micro"
+    ),
+}
+
+
+SKLEARN_METRICS = _SKLEARN_CLASSIFICATION_METRICS.copy()
+SKLEARN_METRICS.update(_SKLEARN_REGRESSION_METRICS)
+
+
+def get_possible_classification_metrics() -> List[str]:
```
Comment (carloszanella) [inline omigami/models/sklearn_metrics.py:99]: small thing: `supported` sounds better to me than `possible`

File: tests/test_feature_selector.py [hunk 2]
```diff
@@ -167,6 +167,9 @@ def test_get_selected_features(fs, mosquito):
     X = mosquito.X[:, 0:10]
     y = np.array([1] + [0, 1] * 14)
     fs.fit(X, y)
+    fs._selected_features = SelectedFeatures(
+        min_feats=[0], max_feats=[0], mid_feats=[0]
+    )
```
Comment (carloszanella) [inline tests/test_feature_selector.py:172]: why did you need to add this? Will break once merged with target branch too due to renaming

File: tests/test_sklearn_metrics.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+import numpy as np
+from omigami.models import sklearn_metrics
+
+
+regression_true = np.array([1, 2, 3, 4, 5])
+regression_pred = np.array([5, 4, 3, 2, 1])
+classification_true = np.array([1, 0, 1, 2, 0, 0])
+classification_pred = np.array([2, 0, 1, 1, 0, 1])
+binary_classification_true = np.array([1, 0, 1, 1, 0, 0])
+binary_classification_pred = np.array([1, 0, 1, 0, 1, 1])
```
Comment (carloszanella) [inline tests/test_sklearn_metrics.py:11]: cool tests

File: tests/test_sklearn_metrics.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+def test_neg_mean_squared_error(y_pred, y_true, res):
+    metric = sklearn_metrics.SKLEARN_METRICS["neg_mean_squared_error"]
+    score = metric(y_true, y_pred)
+    assert res == score
+
+
+@pytest.mark.parametrize(
+    ("y_pred", "y_true", "res"),
+    [
+        (regression_pred, regression_true, -0.587156711483398),
```
Comment (carloszanella) [inline tests/test_sklearn_metrics.py:80]: can't you set res here like you did in the test below, calling a numpy operation?

File: omigami/models/metrics.py [hunk 5]
```diff
@@ -17,9 +17,8 @@ def make_metric(metric: Union[str, MetricFunction]) -> MetricFunction:
 def _make_metric_from_string(metric_string: str) -> MetricFunction:
     if metric_string == "MISS":
         return miss_score
-    if metric_string in SCORERS:
-        # pylint: disable=protected-access
-        return get_scorer(metric_string)._score_func
+    if metric_string in SKLEARN_METRICS:
+        return SKLEARN_METRICS[metric_string]
```
Comment (carloszanella) [inline omigami/models/metrics.py:21]: looks nicer now

**Review by carloszanella — COMMENTED** [review]
I'm curious to see how the linear regression looks like now - the one I sent the plot that led us to finding this issue

File: omigami/models/sklearn_metrics.py (hunk 1 above)
Comment (carloszanella) [reply omigami/models/sklearn_metrics.py:99]: I didn't see this being used elsewhere though

Comment (giulio-tirabassi) [reply omigami/models/sklearn_metrics.py:99]: I wanted to mention it in the readme instead than list all of them

File: tests/test_feature_selector.py (hunk 2 above)
Comment (giulio-tirabassi) [reply tests/test_feature_selector.py:172]: right, just to control which are the selected features, this way it's easier to test

Comment (giulio-tirabassi) [reply tests/test_feature_selector.py:172]: I'll rewrite it to be consistent with the naming (and merge the target branch in to run the tests)

File: tests/test_sklearn_metrics.py (hunk 4 above)
Comment (giulio-tirabassi) [reply tests/test_sklearn_metrics.py:80]: :+1: I will, it's just a complex expression I thought it was clearer to put the number :)

**Comment by giulio-tirabassi** [comment]
for your curiosity: 😄 
This is the plot for an SVR, linear kernel, fitting the normalized logarithms. The score is the RMSE.
immagine
This is a the same but for linear regression. As you see it still goes wild, but at least the features are selected at the minimum
immagine

**Review by carloszanella — APPROVED** [review]

**Comment by giulio-tirabassi** [comment]
@carloszanella I didn't trust that spike too much and I checked deeper: It was a bug in the end 🐛  
It was just for pipelines though. I'll push the fix and a test against it in this same PR. Here you have the plot with the correction. Way more reasonable.
immagine
